### PR TITLE
Fix x check --stage 1 when download-ci-llvm=false

### PR DESCRIPTION
**Pull request by jyn514**
Bootstrap tries to avoid building LLVM unless it needs to; in particular we only build it for `x build`, not `x check`. Unfortunately, the check forgot about existence of stages - it would break if you used `x check --stage 1`:
```
  = note: /usr/bin/ld: cannot find -lPolly: No such file or directory
          /usr/bin/ld: cannot find -lPollyISL: No such file or directory
```
Fix it to work for stage 1.

I recommend reading this commit-by-commit; the first one makes a bunch of whitespace changes but otherwise doesn't change the logic.

### CODE REVIEW

File: src/bootstrap/compile.rs [hunk 1]
```diff
@@ -867,83 +872,84 @@ pub fn rustc_cargo_env(builder: &Builder<'_>, cargo: &mut Cargo, target: TargetS
...
-        if let Some(s) = target_config.and_then(|c| c.llvm_config.as_ref()) {
-            cargo.env("CFG_LLVM_ROOT", s);
-        }
+    let building_is_expensive = crate::llvm::prebuilt_llvm_config(builder, target).is_err();
+    // `top_stage == stage` might be false for `check --stage 1`, if we are building the stage 1 compiler
+    let can_skip_build = builder.kind == Kind::Check && builder.top_stage == stage;
+    let should_skip_build = building_is_expensive && can_skip_build;
+    if builder.config.llvm_enabled() && !should_skip_build {
+        rustc_llvm_env(builder, cargo, target)
+    }
```
Comment (ozkanonur) [inline src/bootstrap/compile.rs:885]: I would prefer running `prebuilt_llvm_config` only if llvm is enabled. Not a big deal, maybe it's just me:)

```suggestion
    if builder.config.llvm_enabled() {
        let building_is_expensive = crate::llvm::prebuilt_llvm_config(builder, target).is_err();
        // `top_stage == stage` might be false for `check --stage 1`, if we are building the stage 1 compiler
        let can_skip_build = builder.kind == Kind::Check && builder.top_stage == stage;
        let should_skip_build = building_is_expensive && can_skip_build;

        if !should_skip_build {
            rustc_llvm_env(builder, cargo, target)
        }
    }
```

**Review by onur-ozkan — APPROVED** [review]
I like you seperated good amount of code block into `rustc_llvm_env`.

I have one non-blocker suggestion, other than that LGTM. Feel free to queue the PR with r=me. Taking or ignoring the suggestion is up to you since it's not a blocker.

File: src/bootstrap/compile.rs (hunk 1 above)
Comment (jyn514) [reply src/bootstrap/compile.rs]: good idea, done :)

**Pull request merged**